.github/workflows/ci.yaml

```diff
@@ -0,0 +1,82 @@
+  pull_request:
+
+name: Continuous integration
+
+jobs:
+  ci-linux:
+    runs-on: ubuntu-20.04
+    continue-on-error: ${{ matrix.experimental || false }}
+    strategy:
+      matrix:
+        # All generated code should be running on stable now, MRSV is 1.42.0
+        rust: [nightly, stable, 1.42.0]
+
+        include:
+          # Nightly is only for reference and allowed to fail
+          - rust: nightly
+            experimental: true
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          override: true
+      - name: Install all Rust targets for ${{ matrix.rust }}
+        run: rustup target install --toolchain=${{ matrix.rust }} x86_64-unknown-linux-gnu riscv32imac-unknown-none-elf riscv64imac-unknown-none-elf riscv64gc-unknown-none-elf
+      - name: Run CI script for x86_64-unknown-linux-gnu under ${{ matrix.rust }}
+        run: |
+          cargo check --target x86_64-unknown-linux-gnu --features board-hifive1
+          cargo check --target x86_64-unknown-linux-gnu --features board-hifive1-revb
+          cargo check --target x86_64-unknown-linux-gnu --features board-redv
+          cargo check --target x86_64-unknown-linux-gnu --features board-lofive
+          cargo check --target x86_64-unknown-linux-gnu --features board-lofive-r1
+      - name: Run CI script for riscv32imac-unknown-none-elf under ${{ matrix.rust }}
+        run: |
+          cargo check --target riscv32imac-unknown-none-elf --features board-hifive1
+          cargo check --target riscv32imac-unknown-none-elf --features board-hifive1-revb
+          cargo check --target riscv32imac-unknown-none-elf --features board-redv
+          cargo check --target riscv32imac-unknown-none-elf --features board-lofive
+          cargo check --target riscv32imac-unknown-none-elf --features board-lofive-r1
+      - name: Run CI script for riscv64imac-unknown-none-elf under ${{ matrix.rust }}
+        run: |
+          cargo check --target riscv64imac-unknown-none-elf --features board-hifive1
+          cargo check --target riscv64imac-unknown-none-elf --features board-hifive1-revb
+          cargo check --target riscv64imac-unknown-none-elf --features board-redv
+          cargo check --target riscv64imac-unknown-none-elf --features board-lofive
+          cargo check --target riscv64imac-unknown-none-elf --features board-lofive-r1
+      - name: Run CI script for riscv64gc-unknown-none-elf under ${{ matrix.rust }}
+        run: |
+          cargo check --target riscv64gc-unknown-none-elf --features board-hifive1
+          cargo check --target riscv64gc-unknown-none-elf --features board-hifive1-revb
+          cargo check --target riscv64gc-unknown-none-elf --features board-redv
+          cargo check --target riscv64gc-unknown-none-elf --features board-lofive
+          cargo check --target riscv64gc-unknown-none-elf --features board-lofive-r1
```

Comment:
Hmm, are you sure we need this? The board is RV32IMAC.

Reply:
Same for e310x/e310x-hal

Reply:
fixed